tests/h/paginator_test.py

```diff
@@ -1,158 +1,199 @@
+        ('1', None),
+    ])
+    def test_prev(self, pyramid_request, page_param, expected):
+        pyramid_request.params = {'page': page_param}
+        assert paginate(pyramid_request, 600, 10)['prev'] == expected
+
+
+@pytest.mark.usefixtures('paginate')
+class TestPaginateQuery(object):
+
+    CURRENT_PAGE = 3
+    "The current page that will be returned by paginate()."
+
+    PAGE_SIZE = 10
+    "The page_size argument that will be passed to paginate_query()."
+
+    def test_it_calls_the_wrapped_view_callable(self,
+                                                pyramid_request,
+                                                view_callable,
+                                                wrapped):
+        """It calls the wrapped view callable to get the SQLAlchemy query."""
+        wrapped(mock.sentinel.context, pyramid_request)
+
+        view_callable.assert_called_once_with(mock.sentinel.context,
+                                              pyramid_request)
+
+    def test_it_calls_paginate(self,
+                               paginate,
+                               pyramid_request,
+                               wrapped):
+        """It calls paginate() to get the paginator template data."""
+        wrapped(mock.sentinel.context, pyramid_request)
+
+        paginate.assert_called_once_with(pyramid_request,
+                                         mock.sentinel.total,
+                                         self.PAGE_SIZE)
+
+    def test_it_offsets_the_query(self,
+                                  wrapped,
+                                  pyramid_request,
+                                  query):
+        """
+        It offsets the query by the correct amount.
+
+        It offsets the query so that only the results starting from the first
+        result that should be shown on the current page (as returned by
+        paginate()) are fetched from the db.
+
+        """
+        wrapped(mock.sentinel.context, pyramid_request)
+
+        # The current page is 3, and there are 10 results per page, so we
+        # would expect the 20 first results (the first two pages) to be offset.
+        query.offset.assert_called_once_with(20)
+
+    def test_it_limits_the_query(self,
+                                 wrapped,
+                                 pyramid_request,
+                                 query):
+        """
+        It limits the query by the correct amount.
+
+        It limits the query so that only the results up to the last result that
+        should be shown on the current page are fetched from the db.
+
+        """
+        wrapped(mock.sentinel.context, pyramid_request)
+
+        query.limit.assert_called_once_with(self.PAGE_SIZE)
+
+    def test_it_returns_the_query_results(self,
+                                          wrapped,
+                                          pyramid_request):
+        assert wrapped(mock.sentinel.context, pyramid_request)['results'] == (
```

Comment:
If your assertion is split over two lines like this, perhaps it would be better to write:

```python
results = wrapped(mock.sentinel.context, pyramid_request)['results']

assert results == mock.sentinel.all
```